flight/PiOS/STM32/pios_debug.c

```diff
@@ -153,6 +157,15 @@ void PIOS_DEBUG_PinValue4BitL(uint8_t value)
  */
 void PIOS_DEBUG_Panic(const char *msg)
 {
+#ifdef PIOS_INCLUDE_RTOS
+	/* Stop all other threads (once we get to the infloop, at least)
+	 * by elevating our priority.  HIGHPRIO is a chibios constant,
+	 * so if we change RTOS we'll know this will need to change by
+	 * compile error.
+	 */
+	PIOS_Thread_ChangePriority(HIGHPRIO);
```

Comment:
Did you consider chSysHalt (not sure if our old rel has it)? http://chibios.sourceforge.net/docs3/rt/group__system.html#gaffab4052111407f092c7c65466a61450

Reply:
We could do that.  Couldn't do it this early in case PIOS_COM blocks.  I dunno, it seems like a wash.

Reply:
(Or even just PIOS_IRQ_Disable...)

Reply:
PIOS_IRQ_Disable seems like a good idea. That will kill RTOS + interrupt handlers in one fell swoop right?

Reply:
Yes, and it will even work on flyingpio, so will do that.